bookkeeper-common-allocator/src/main/java/org/apache/bookkeeper/common/allocator/ByteBufAllocatorBuilder.java

```diff
@@ -39,7 +39,7 @@ static ByteBufAllocatorBuilder create() {
     /**
      * Finalize the configured {@link ByteBufAllocator}.
      */
-    ByteBufAllocator build();
+    ByteBufAllocatorWithOomHandler build();
```

Comment:
is this used by Pulsar ?
Probably it will cause some NoSuchMethodError for users who try to upgrade BookKeeper without upgrading Pulsar (I know some projects that create a new Pulsar package and select different BookKeeper versions)

is it necessary to change this ?

Reply:
can we move this ByteBufAllocator change to a separate PR ?

Reply:
I don't know what you are referring to regarding Pulsar. 

But regarding a separate PR, this is related to this construction/injection PR as we allow the OOM handling to be set in a different place to the construction of the allocator. Basically it gets constructed, then later the BookieService sets the OOM handler so that it integrates with the AbstractLifecycleComponent uncaughtExceptionHandler which will trigger the bookie to shutdown gracefully on an OOM.

Reply:
ok, let's keep this.

for Pulsar, IIRC we recently started to use BK ByteBufAllocator to deal with OutOfMemory cases for Direct Memory
and we are using this library